solutions/cses-1090.mdx

```diff
@@ -84,48 +85,44 @@ int main(){
 
 <JavaSection>
 
-<Warning>
-Java will TLE on test case #10.
-</Warning>
-
 ```java
-import java.io.*;
 import java.util.*;
 
 public class FerrisWheel {
-    public static void main(String[] args) {
-        Kattio io = new Kattio();
-
-        int n = io.nextInt();
-        int x = io.nextInt();
-
-        int[] weights = new int[n];
-        for (int i = 0; i < n; i++) {
-            weights[i] = io.nextInt();
-        }
-        Arrays.sort(weights);
-
-        int ans = 0;
-        int i = 0;  // left pointer
-        int j = n - 1;  // right pointer
-        while (i <= j) {
-            ans++;  // Increment number of gondolas used
-            if (i == j) break;
-            // If total weight is too large, move to lighter child
-            if (weights[i] + weights[j] > x) {
-                j--;
-            }
-            // Otherwise, we pair the two children and move on
-            else {
-                i++;
-                j--;
-            }
-        }
-        io.println(ans);
-        io.close();
-    }
-
-    // CodeSnip{Kattio}
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
+
+		int n = sc.nextInt();
+		int x = sc.nextInt();
+		sc.nextLine();
+
+		// Read in weights of children (this optimization is necessary to pass 
+		// all test cases)
+		String[] weightsStr = sc.nextLine().split(" ");
+		Integer[] weights = new Integer[n];
```

Comment:
is using `Integer` really necessary

Reply:
yes, if I don't it would TLE because Integer.parseInt() returns an Integer and casting it to an int takes too long

Reply:
maybe add that as a comment to the code